### PR TITLE
DR-2427 Return Flight className in FlightState object

### PR DESCRIPTION
**Pull request by nmalfroy**
Fairly simple PR that just adds the className to the FlightState object (that data was already stored in the DB)

### CODE REVIEW

File: stairway/src/test/java/bio/terra/stairway/FlightStateClassTest.java
Comment (ddietterich) [inline stairway/src/test/java/bio/terra/stairway/FlightStateClassTest.java]: This could be `private static final`

**Review by ddietterich — APPROVED** [review]
LGTM - one nit

File: stairway/src/test/java/bio/terra/stairway/FlightStateClassTest.java
Comment (nmalfroy) [reply stairway/src/test/java/bio/terra/stairway/FlightStateClassTest.java]: ah good call.  Update that and the `bad` var to be static final

**Comment by nmalfroy** [comment]
Gah, sorry, Dan, didn't mean to re-add you as a reviewer

**Review by ddietterich — APPROVED** [review]
Make sure you run spotlessApply before your final merge. Looks like there are some back-and-forth changes that look like spotless

**Comment by jmczerk** [comment]
Just a heads up that I am OOO til Friday, so don't feel a need to wait on my review if you get two approvals before then.

**Comment by nmalfroy** [comment]
Ah good point.  I thought I had but I'll update that.  I'll see about getting another review @jmczerk , thanks!

**Comment by nmalfroy** [comment]
I ran spotless and it looks ok.  I think the back and forth might be from uppercasing the static final

**Comment by ddietterich** [comment]
> 

Thank you

File: stairway/src/test/java/bio/terra/stairway/impl/EnumerateFlightsTest.java
Comment (pshapiro4broad) [inline stairway/src/test/java/bio/terra/stairway/impl/EnumerateFlightsTest.java:159]: It bugs me that there's no better way to do this. It seems like there should be a way to stream both lists at the same time and assertThat() on each pair of elements in order, avoiding the need to create a collection.

File: stairway/src/test/java/bio/terra/stairway/impl/EnumerateFlightsTest.java
Comment (pshapiro4broad) [inline stairway/src/test/java/bio/terra/stairway/impl/EnumerateFlightsTest.java:171]: It looks like there's a common pattern here that could be abstracted, if you wanted to. The abstraction is the getter on the object, so the general method is something like 
```
assertSameField(List<FlightState> expected, List<FlightState> actual, Function<FlightState, T> getter)
```

File: stairway/src/test/java/bio/terra/stairway/impl/FilterTest.java
Comment (pshapiro4broad) [inline stairway/src/test/java/bio/terra/stairway/impl/FilterTest.java:221]: IMO these kind of string comparison tests aren't very useful and the code is better served by tests that run the SQL. That being said, I would at least create strings for the common substrings to avoid needless code duplication.

**Review by pshapiro4broad — APPROVED** [review]
Change looks fine, some comments on the DRY-ness of the test code.

File: stairway/src/test/java/bio/terra/stairway/impl/FilterTest.java
Comment (ddietterich) [reply stairway/src/test/java/bio/terra/stairway/impl/FilterTest.java:221]: The main complexity in the filtering code is generating the right SQL. This kind of test makes it much more direct to ensure that is working, rather than debugging SQL errors inside the larger system.

File: stairway/src/test/java/bio/terra/stairway/impl/EnumerateFlightsTest.java
Comment (nmalfroy) [reply stairway/src/test/java/bio/terra/stairway/impl/EnumerateFlightsTest.java:159]: I don't know...I guess this doesn't bug me that much...My bigger issue is that I just don't like the JUnit matchers...The assertJ matchers have more explicit list checkers

File: stairway/src/test/java/bio/terra/stairway/impl/EnumerateFlightsTest.java
Comment (nmalfroy) [reply stairway/src/test/java/bio/terra/stairway/impl/EnumerateFlightsTest.java:171]: Yeah, maybe...though they are subtly different so it may not be super commonizable

Comment (pshapiro4broad) [reply stairway/src/test/java/bio/terra/stairway/impl/EnumerateFlightsTest.java:171]: If the comparisons are different, you can write
```
assertSameField(List<FlightState> expected, List<FlightState> actual, 
                             Function<FlightState, T> getter, Matcher<T>)
```

**Pull request merged**
